src/plugins/intel_gpu/tests/unit/shape_infer/matmul_si_test.cpp

```diff
@@ -110,7 +110,7 @@ INSTANTIATE_TEST_SUITE_P(smoke, fully_connected_test,
         {
             layout{ov::PartialShape{10, 1024}, data_types::f32, format::bfyx},
             layout{ov::PartialShape{1000, 1024}, data_types::f32, format::bfyx},
-            data_types::i32, false, false,
+            data_types::f32, false, false,
```

Comment:
Why don't we add a new test instead of fixing existing one?

Reply:
For the above reason, previously even if the output data type not supported by FC, such as i32, was specified in prim, the input data type was propagated internally during layout calculation. But now the behavior related to layout calculation for FC has changed, so this fix is necessary.